pkg/server/response.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/xml"
+	"net/http"
+
+	"github.com/wzshiming/s3d/pkg/s3types"
+)
+
+// xmlResponse writes an XML response
+func (s *S3Server) xmlResponse(w http.ResponseWriter, data interface{}, status int) {
+	w.Header().Set("Content-Type", "application/xml")
+	w.WriteHeader(status)
+
+	w.Write([]byte(xml.Header))
+	xml.NewEncoder(w).Encode(data)
```

Comment:
Error return value of `(*encoding/xml.Encoder).Encode` is not checked (errcheck)